cmd/juju/application/deployer/bundlehandler.go

```diff
@@ -1711,3 +1723,13 @@ func addCharmConstraintsParser(s string) bundlechanges.ArchConstraint {
 		constraints: s,
 	}
 }
+
+func verifyEndpointBindings(endpointBindings map[string]string, knownSpaces []string) error {
+	knownSpacesSet := set.NewStrings(knownSpaces...)
+	for _, spaceName := range endpointBindings {
+		if !knownSpacesSet.Contains(spaceName) {
+			return errors.NotFoundf("space %q", spaceName)
```

Comment:
`NotValidf` might be suitable too, but I think this is OK.

Reply:
`NotFoundf` is used in many places elsewhere, so I think I'll stick with it